In [17]:
import pandas as pd

In [18]:
df = pd.read_csv('netflix_titles1.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6236 entries, 0 to 6235
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   duration_minutes  4267 non-null   object 
 1   duration_seasons  1971 non-null   object 
 2   type              6235 non-null   object 
 3   title             6235 non-null   object 
 4   date_added        6223 non-null   object 
 5   release_year      6234 non-null   float64
 6   rating            6223 non-null   object 
 7   description       6233 non-null   object 
 8   show_id           6232 non-null   float64
dtypes: float64(2), object(7)
memory usage: 438.6+ KB


In [19]:
df[df['show_id'].isna()]

duration_minutes duration_seasons   type  \
2017                    40              NaN  Movie   
2018      Flying Fortress"        3/31/2017   1944   
4524                    87              NaN  Movie   
4525   and probably will."         80188902    NaN   

                                title  \
2017  The Memphis Belle: A Story of a   
2018                            TV-PG   
4524          The Bad Education Movie   
4525                              NaN   

                                             date_added  release_year rating  \
2017                                                NaN           NaN    NaN   
2018  This documentary centers on the crew of the B-...    80119194.0    NaN   
4524                                         12/15/2018        2015.0  TV-MA   
4525                                                NaN           NaN    NaN   

                                            description  show_id  
2017                                                NaN      NaN  
2018                                                NaN      NaN  
4524  Britain's most ineffective but caring teacher ...      NaN  
4525                                                NaN      NaN

Data yang ada di row ke 2017 dan 4524 sepertinya mengalami pemisahan menjadi dua row yang berbeda karena show_id sebagai primary key seharusnya tidak boleh NaN. Oleh karena itu, saya akan merubah nilai dari kedua kolom tersebut di file Excel agar datanya menjadi konsisten.

Data yang sudah diperbaiki saya simpan di folder fixed dengan nama 'netflix_titles.xlsx'. Sementara itu, Excel workbook yang lama saya ganti namanya menjadi 'old_netflix_titles.xlsx'.

In [20]:
df_title = pd.read_csv('fixed/netflix_titles2.csv')
df_title.info()

duplicate_count_column = df_title['show_id'].duplicated().sum()
print(duplicate_count_column)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6234 entries, 0 to 6233
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   duration_minutes  4265 non-null   float64
 1   duration_seasons  1969 non-null   float64
 2   type              6234 non-null   object 
 3   title             6234 non-null   object 
 4   date_added        6223 non-null   object 
 5   release_year      6234 non-null   int64  
 6   rating            6224 non-null   object 
 7   description       6234 non-null   object 
 8   show_id           6234 non-null   int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 438.5+ KB
0


In [21]:
df_director = pd.read_csv('fixed/netflix_titles_directors.csv')
df_director.info()
duplicate_count_column = df_director['show_id'].duplicated().sum()
print(duplicate_count_column)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4852 entries, 0 to 4851
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   director  4852 non-null   object
 1   show_id   4852 non-null   int64 
dtypes: int64(1), object(1)
memory usage: 75.9+ KB
587


In [22]:
df_country = pd.read_csv('fixed/netflix_titles_countries.csv')
df_country.info()
duplicate_count_column = df_country['show_id'].duplicated().sum()
print(duplicate_count_column)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7179 entries, 0 to 7178
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   country  7179 non-null   object
 1   show_id  7179 non-null   int64 
dtypes: int64(1), object(1)
memory usage: 112.3+ KB
1421


In [23]:
df_cast = pd.read_csv('fixed/netflix_titles_cast.csv')
df_cast.info()
duplicate_count_column = df_cast['show_id'].duplicated().sum()
print(duplicate_count_column)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44311 entries, 0 to 44310
Data columns (total 2 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   cast     44311 non-null  object
 1   show_id  44311 non-null  int64 
dtypes: int64(1), object(1)
memory usage: 692.5+ KB
38647


In [24]:
df_category = pd.read_csv('fixed/netflix_titles_category.csv')
df_category.info()
duplicate_count_column = df_category['show_id'].duplicated().sum()
print(duplicate_count_column)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13670 entries, 0 to 13669
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   listed_in  13670 non-null  object
 1   show_id    13670 non-null  int64 
dtypes: int64(1), object(1)
memory usage: 213.7+ KB
7436
